index.js

```diff
@@ -1,62 +1,109 @@
 'use strict';
 
+// Built-in types
 var types = [
   'object',
   'number',
   'string',
   'symbol',
   'boolean',
+  'date',
   'function', // Weird to expose this
-  'undefined',
+  'undefined', // And this?
 ];
 
-function normalize(type, value) {
-  var args = Array.prototype.slice.call(arguments, 2);
 
-  var isType = conforms(type, value);
-  var isFunction = typeof value === 'function';
+function normalize(coercer, value) {
+  if (typeof value === 'function') {
+    if (coercer === 'function') {
+      return value;
+    }
+    value = value.apply(null, Array.prototype.slice.call(arguments, 2));
+  }
+  return coerce(coercer, value);
+}
+
+
+function coerce(coercer, value) {
 
-  if (!isType && !isFunction) {
-    return null;
+  // Handle built-in types
+  if (typeof coercer === 'string') {
+    if (coerce[coercer]) {
+      return coerce[coercer](value);
+    }
+    return fallback(coercer, value);
   }
 
-  if (isType) {
-    return value;
+  // Handle custom coercer
+  if (typeof coercer === 'function') {
+    return coercer(value);
   }
 
-  var result = value.apply(null, args);
+  // Array of coercers, try in order until one returns a non-null value
+  var result = null;
+  coercer.some(function(coercer) {
+    result = coerce(coercer, value);
+    return result !== null;
+  });
+
+  return result;
+}
+
+
+coerce.string = function(value) {
+  value = primitive(value);
+  if (value != null &&
+    typeof value === 'object' &&
+    typeof value.toString === 'function') {
 
-  if (conforms(type, result)) {
-    return result;
+    value = value.toString();
   }
+  return fallback('string', value);
+};
+
+
+coerce.number = function(value) {
+  return fallback('number', primitive(value));
+};
 
+
+coerce.boolean = function(value) {
+  return fallback('boolean', primitive(value));
+};
+
+
+coerce.date = function(value) {
+  value = primitive(value);
+  if (typeof value === 'number' && !isNaN(value) && isFinite(value)) {
+    return new Date(value);
+  }
   return null;
-}
+};
+
 
-function conforms(type, value) {
-  if (typeof type === 'string') {
-    return typeof value === type;
+function fallback(type, value) {
```

Comment:
maybe just name it `typeOf` since that's what it does and how it is used.  `typeOf('string', value)`